mule-extensions-soap-api/src/main/java/org/mule/runtime/extension/api/soap/message/DispatcherResponse.java

```diff
@@ -18,10 +20,12 @@
 
   private final String contentType;
   private final InputStream content;
+  private final Map<String, ? extends List<String>> headers;
 
-  public DispatcherResponse(String contentType, InputStream content) {
+  public DispatcherResponse(String contentType, InputStream content, Map<String, ? extends List<String>> headers) {
```

Comment:
? extends List -> List. ALso, response object should have Optional<List<String>> getHeaders(String)